modules/Individual Needs/investigations_submit_detailProcess.php

```diff
@@ -0,0 +1,132 @@
+$gibbonINInvestigationContributionID = $_POST['gibbonINInvestigationContributionID'] ?? '';
+
+$URL = $_SESSION[$guid]['absoluteURL']."/index.php?q=/modules/Individual Needs/investigations_submit_detail.php&gibbonINInvestigationID=$gibbonINInvestigationID&gibbonINInvestigationContributionID=$gibbonINInvestigationContributionID";
+$URLSuccess = $_SESSION[$guid]['absoluteURL']."/index.php?q=/modules/Individual Needs/investigations_submit.php&gibbonINInvestigationID=$gibbonINInvestigationID&gibbonINInvestigationContributionID=$gibbonINInvestigationContributionID";
+
+if (isActionAccessible($guid, $connection2, '/modules/Individual Needs/investigations_submit_detail.php') == false) {
+    $URL .= '&return=error0';
+    header("Location: {$URL}");
+    exit;
+} else {
+    // Validate the database records exist
+    $investigationGateway = $container->get(INInvestigationGateway::class);
+    $criteria = $investigationGateway->newQueryCriteria();
+    $investigation = $investigationGateway->queryInvestigationsByID($criteria, $gibbonINInvestigationID, $_SESSION[$guid]['gibbonSchoolYearID']);
+    $investigation = $investigation->getRow(0);
+
+    $contributionsGateway = $container->get(INInvestigationContributionGateway::class);
+    $criteria2 = $contributionsGateway->newQueryCriteria();
+    $contribution = $contributionsGateway->queryContributionsByID($criteria2, $gibbonINInvestigationContributionID);
+    $contribution = $contribution->getRow(0);
+
+    if (empty($investigation) || empty($contribution) || $contribution['gibbonPersonID'] != $gibbon->session->get('gibbonPersonID')) {
+        $URL .= '&return=error0';
+        header("Location: {$URL}");
+        exit;
+    } else {
+        $data = [
+            'status'            => 'Complete',
+            'cognition'         => $_POST['cognition'] ?? null,
+            'memory'            => (!empty($_POST['memory'])) ? serialize($_POST['memory']) : '',
+            'selfManagement'    => (!empty($_POST['selfManagement'])) ? serialize($_POST['selfManagement']) : '',
+            'attention'         => (!empty($_POST['attention'])) ? serialize($_POST['attention']) : '',
+            'socialInteraction' => (!empty($_POST['socialInteraction'])) ? serialize($_POST['socialInteraction']) : '',
+            'communication'     => (!empty($_POST['communication'])) ? serialize($_POST['communication']) : '',
+            'comment'           => $_POST['comment'] ?? ''
+        ];
+
+        // Validate the required values are present
+        if (empty($data['cognition'])) {
+            $URL .= '&return=error1';
+            header("Location: {$URL}");
+            exit;
+        }
+
+        // Update the record
+        $updated = $contributionsGateway->update($gibbonINInvestigationContributionID, $data);
+
+        //Check for completion, update status and issue notifications
+        $completion = $contributionsGateway->queryInvestigationCompletion($criteria2, $gibbonINInvestigationID);
```

Comment:
Btw, not everything has to be a query with a $criteria object, those are designed in particular to be used with the DataTables. A `selectInvestigationCompletion()` might be simpler here.

Reply:
I was wondering about that. So, selectInvestigationCompletion is something I'd have to write?

Reply:
Yep. I'll tweak some stuff in post-merge :)